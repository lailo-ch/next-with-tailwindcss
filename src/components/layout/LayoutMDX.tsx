import * as React from 'react'
import { MDXProvider } from '@mdx-js/react'
import Layout, { Props as LayoutProps } from './Layout'
import Text from '@components/Text'
import List from '@components/List'

const components = {
  h1: ({ children }) => (
    <Text h1 className="mb-4">
      {children}
    </Text>
  ),
  h2: ({ children }) => (
    <Text h2 className="mb-4">
      {children}
    </Text>
  ),
  h3: ({ children }) => (
    <Text h3 className="mb-4">
      {children}
    </Text>
  ),
  h4: ({ children }) => (
    <Text h4 className="mb-4">
      {children}
    </Text>
  ),
  h5: ({ children }) => (
    <Text h5 className="mb-4">
      {children}
    </Text>
  ),
  h6: ({ children }) => (
    <Text h6 className="mb-4">
      {children}
    </Text>
  ),
  p: ({ children }) => <Text className="mb-4">{children}</Text>,
  strong: ({ children }) => (
    <Text bold className="mb-4">
      {children}
    </Text>
  ),
  sub: ({ children }) => (
    <Text small className="mb-4">
      {children}
    </Text>
  ),
  ul: ({ children }) => <List className="mb-4">{children}</List>,
  ol: ({ children }) => (
    <List ordered className="mb-4">
      {children}
    </List>
  ),
  li: ({ children }) => <List item>{children}</List>,
}

const LayoutMDX: React.FC<LayoutProps> = ({ children, ...props }) => {
  return (
    <Layout {...props}>
      <MDXProvider components={components}>{children}</MDXProvider>
    </Layout>
  )
}

export default LayoutMDX
