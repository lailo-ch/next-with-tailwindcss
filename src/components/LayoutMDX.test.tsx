import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LayoutMDX from './LayoutMDX'

describe('LayoutMDX', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<LayoutMDX>hello world</LayoutMDX>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
