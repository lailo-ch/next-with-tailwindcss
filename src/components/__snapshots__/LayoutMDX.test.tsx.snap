// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutMDX matches snapshot 1`] = `
<Layout>
  <MDXProvider
    components={
      Object {
        "h1": [Function],
        "h2": [Function],
        "h3": [Function],
        "h4": [Function],
        "h5": [Function],
        "h6": [Function],
        "li": [Function],
        "ol": [Function],
        "p": [Function],
        "strong": [Function],
        "sub": [Function],
        "ul": [Function],
      }
    }
  >
    hello world
  </MDXProvider>
</Layout>
`;
